cmd/tree/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+)
+
+func tree(path string, curDepth int, maxDepth int, numDirs *int, numFiles *int) {
+	if curDepth > maxDepth {
+		return
+	}
+
+	file, err := os.ReadDir(path)
+	if err != nil {
+		log.Fatalf("Error happened: %v", err)
```

Comment:
don't exit from the function instead propagate the error and leave the error handling to the caller